stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -215,23 +218,36 @@ public class MessageListViewModel(
             }
         }
 
+    /**
+     * A background job used for view model initialization.
+     * The job should be canceled after receiving the first, non-null value from the watch channel request.
+     */
+    private var initialJob: Job? = null
+
     init {
-        initWithOfflinePlugin()
+        stateMerger.addSource(MutableLiveData(State.Loading)) { stateMerger.value = it }
+
+        initialJob = viewModelScope.launch {
+            chatClient.watchChannelAsState(
+                cid = cid,
+                messageLimit = DEFAULT_MESSAGES_LIMIT,
+                coroutineScope = viewModelScope
+            ).collect { channelState ->
```

Comment:
Unfortunately no `firstNotNull()` function in the standard library, only `first { it!= null }`.